crates/world-id-core/src/credential.rs

```diff
@@ -0,0 +1,157 @@
+    expires_at: u64,
+    /// These are concrete statements that the issuer attests about the receiver.
+    /// Could be just commitments to data (e.g. passport image) or
+    /// the value directly (e.g. date of birth)
+    #[serde(serialize_with = "ark_serde_compat::serialize_babyjubjub_base_sequence")]
+    #[serde(deserialize_with = "ark_serde_compat::deserialize_babyjubjub_base_sequence")]
+    claims: Vec<BaseField>,
+    /// If needed, can be used as commitment to the underlying data.
+    /// This can be useful to tie multiple proofs about the same data together.
+    #[serde(serialize_with = "ark_serde_compat::serialize_babyjubjub_base")]
+    #[serde(deserialize_with = "ark_serde_compat::deserialize_babyjubjub_base")]
+    associated_data_hash: BaseField,
+}
+
+impl Credential {
+    /// Create a new credential.
+    pub fn new() -> Self {
+        Self {
+            version: CredentialVersion::V1,
+            issuer_schema_id: 0,
+            account_id: 0,
+            genesis_issued_at: 0,
+            expires_at: 0,
+            claims: vec![BaseField::zero(); MAX_CLAIMS],
+            associated_data_hash: BaseField::zero(),
+        }
+    }
+
+    /// Set the version of the credential.
+    pub fn version(mut self, version: CredentialVersion) -> Self {
+        self.version = version;
+        self
+    }
+
+    /// Set the issuer schema id of the credential.
+    pub fn issuer_schema_id(mut self, issuer_schema_id: u64) -> Self {
+        self.issuer_schema_id = issuer_schema_id;
+        self
+    }
+
+    /// Set the account id of the credential.
+    pub fn account_id(mut self, account_id: u64) -> Self {
+        self.account_id = account_id;
+        self
+    }
+
+    /// Set the genesis issued at of the credential.
+    pub fn genesis_issued_at(mut self, genesis_issued_at: u64) -> Self {
+        self.genesis_issued_at = genesis_issued_at;
+        self
+    }
+
+    /// Set the expires at of the credential.
+    pub fn expires_at(mut self, expires_at: u64) -> Self {
+        self.expires_at = expires_at;
+        self
+    }
+
+    /// Set the claim of the credential.
+    pub fn claim(mut self, index: usize, claim: U256) -> Result<Self, anyhow::Error> {
+        if index >= self.claims.len() {
+            return Err(anyhow::anyhow!("Index of claim out of bounds"));
+        }
+        self.claims[index] = claim.try_into()?;
+        Ok(self)
+    }
+
+    /// Set the associated data hash of the credential.
+    pub fn associated_data_hash(
+        mut self,
+        associated_data_hash: U256,
+    ) -> Result<Self, anyhow::Error> {
+        self.associated_data_hash = associated_data_hash.try_into()?;
+        Ok(self)
+    }
+
+    /// Get the credential domain separator for the given version.
+    pub fn get_cred_ds(&self) -> BaseField {
+        match self.version {
+            CredentialVersion::V1 => {
+                BaseField::from_be_bytes_mod_order(b"POSEIDON2+EDDSA-BJJ+DLBE-v1")
```

Comment:
```suggestion
                BaseField::from_be_bytes_mod_order(b"POSEIDON2+EDDSA-BJJ")
```

not sure if want to add +CBOR at the end